File: backend/backend.Application/Models/AppointmentInDTO.cs
using backend.Domain.Enums;

namespace backend.Application.Models
{
    public class AppointmentInDTO
    {
        public int PatientId { get; set; }
        public int DoctorId { get; set; }
        public DateTime AppointmentDate { get; set; }
        public string Diagnosis { get; set; } = string.Empty;
        public string Notes { get; set; } = string.Empty;
        public VisitType VisitType { get; set; }
        public List<PrescriptionDetailInDTO> PrescriptionDetails { get; set; } = new();
    }
}

File: client/src/models/appointmentInDTO.ts
import { PrescriptionDetailInDTO } from "./prescriptionDetailInDTO";
import { VisitType } from "./visitType";

export interface AppointmentInDTO {
  patientId: number;
  doctorId: number;
  appointmentDate: string;
  diagnosis: string;
  notes: string;
  visitType: VisitType;
  prescriptionDetails: PrescriptionDetailInDTO[];
}